### PR TITLE
abbreviations

### PR DESCRIPTION
**Pull request by nschneid**
For uncommon abbreviations, expand to the canonical spelling in the lemma field and specify `Abbr=Yes` (see UniversalDependencies/docs#513). Low-frequency types of (alleged) prepositions, determiners, and conjunctions have been manually inspected and corrected.

### CODE REVIEW

File: not-to-release/sources/answers/20111106215236AAycANO_ans.xml.conllu [hunk 1]
```diff
@@ -333,7 +333,7 @@
 # sent_id = answers-20111106215236AAycANO_ans-0024
 # text = Monday thru Friday
 1	Monday	Monday	PROPN	NNP	Number=Sing	0	root	0:root	_
-2	thru	thru	ADP	IN	_	3	case	3:case	_
+2	thru	through	ADP	IN	Abbr=Yes	3	case	3:case	_
```
Comment (sebschu) [inline not-to-release/sources/answers/20111106215236AAycANO_ans.xml.conllu]: I agree that we should use the canonical spelling for the lemma but does this really count as an abbreviation?

Comment (sebschu) [reply not-to-release/sources/answers/20111106215236AAycANO_ans.xml.conllu]: Okay, now that I looked at the other examples, it probably makes sense to treat this as an abbreviation that is used in texting etc.

**Comment by sebschu** [comment]
Thanks!!

**Comment by dseddah** [comment]
Hi everyone,
how do you treat abbreviation such as lmfao, TL;DR, etc ?
In our upcoming UD version of the French Social Media Bank we are hesitating between

1-6 TL;DR
1 too
2 long
3 ;
4 did
5 n't
6 read

and
1 TL;DR   too+long+;+did+n't+read  

usually, our POS scheme follows the merge POS schema (as it was done in pre-UD treebanks such as korean trebanks (seijong and kaist) for example, namely no morph analysis at the surface level but complexe POS and lemmas) but this is not consistant with UD. More importantly, with time the compositional nature tends to fade away ("just give me a TL;DR and I'll be ok) so i'll be really inclined to not provide internal structures for those.

I'll really be happy to hear everyone's thought on this.

**Comment by jnivre** [comment]
I would recommend doing neither. Etymology is not part of UD annotation. :) If you really want to preserve this information, use the MISC field.

**Comment by jnivre** [comment]
In general, we have to give up the idea that UD can represent any kind of annotation that you would ever like to add to your data. UD is for morphology and syntax. Use other layers for other stuff.

**Comment by dseddah** [comment]
yeah, but how would you annotate stuff like this ?
chépa tro koi faire ?    (je ne sais pas trop quoi faire/ I don't quite know what to do)
where chépa is the contraction of "je sais pas"/I don't know

that's the same problem with thru --> through. Non standard spelling being somehow normalized.

**Comment by dseddah** [comment]
Actually you're not alone in that view: the first time I presented the Cr#pbank to French linguists, Anne Abeillé asked me why were we spending time analyzing and normalizing those forms. She said "those are just utterances of new French, deal with it".

**Comment by dseddah** [comment]
so for her, chépa will simply be a new invariable verb, synonym of to ignore.

**Comment by amir-zeldes** [comment]
I think for a lot of these cases we need to consider two questions: what are users expecting, and what commonalities would each decision preserve/lose. For chepa, I find it difficult to justify treating it as a single token, even if it was originally spelled together, since the negation in there seems regular (though spelled non-standard). 

Would we also want chkroipas 'I don't think so' to be an inherently negative verb next to a totally unrelated positive form 'chkroi'? It seems like more of a spelling issue, at least for the pas: we could segment che/VERB pa/ADV (and use SpaceAfter=No for che, if it's spelled like that).

The other part is about predictability: there might be other ways of spelling these, but the 'normative' form is the most predictable for users of the data looking to find all instances of the lemma 'savoir'. And indeed, the lemma here is not meant as a normalization to 'sais', it's just meant to list 'savoir' somewhere (possibly in a portmanteau lemma with 'je', or some other solution).

**Comment by dseddah** [comment]
I agree with you, especially since we can encounter also things like "ché pas"..  
But what's really giving me headaches is  over splitting cases like "c t"  where "c" usually stands for "c'est" (this is) but here it must be understood as "c'ét ait" (this'w as)
regardless of what UD should encode or not, this one is tough as it requires a n-m mapping between source tokens (words) and tree tokens (tokens in the ud terminology) and this is not currently possible (and yes, word with space doesn't count here :) )

**Comment by nschneid** [comment]
I've created a dedicated issue for this: please relocate to #516

**Comment by jnivre** [comment]
I am not saying it is easy, and I definitely agree that we should try to cover the basic predicate argument structure as well as we can, but unpacking things like "TL;DR" seems going too far.

**Pull request merged**